packages/types/src/interfaces/contractsAbi/definitions.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright 2017-2020 @polkadot/types authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+/* eslint-disable sort-keys */
+
+const layout = {
+  InkLayoutKey: '[u8; 32]',
+  InkStorageLayout: {
+    _enum: {
+      Cell: 'InkLayoutCell',
+      Hash: 'InkLayoutHash',
+      Array: 'InkLayoutArray',
+      Struct: 'InkLayoutStruct',
+      Enum: 'InkLayoutEnum'
+    }
+  },
+  InkLayoutCell: {
+    key: 'InkLayoutKey',
+    ty: 'MtLookupTypeId'
+  },
+  InkCryptoHasher: {
+    _enum: ['Blake2x256', 'Sha2x256', 'Keccak256']
+  },
+  InkLayoutHash: {
+    offset: 'InkLayoutKey',
+    strategy: 'InkLayoutHashingStrategy',
+    layout: 'InkStorageLayout'
+  },
+  InkLayoutHashingStrategy: {
+    hasher: 'InkCryptoHasher',
+    postfix: 'Vec<u8>',
+    prefix: 'Vec<u8>'
+  },
+  InkLayoutArray: {
+    offset: 'InkLayoutKey',
+    len: 'u32',
+    cellsPerElem: 'u64',
+    layout: 'InkStorageLayout'
+  },
+  InkLayoutStruct: {
+    fields: 'Vec<InkLayoutStructField>'
+  },
+  InkLayoutStructField: {
+    layout: 'InkStorageLayout',
+    name: 'Text'
+  },
+  InkDiscriminant: 'u32',
+  InkLayoutEnum: {
+    dispatchKey: 'InkLayoutKey',
+    variants: 'BTreeMap<InkDiscriminant, InkLayoutStruct>'
+  }
+};
+
+const spec = {
+  InkPath: 'Vec<Text>',
+  InkDisplayName: 'InkPath',
```

Comment:
As above, alphabetical since it will create unneeded conflicts with the impl branch. (These are always tricky since (a) needs to be done manually & (b) no checking since the sorting is switched off)